File: matlab_LLV/AcfMed.m
function Acf = AcfMed(lagS,win,TS)
% lagS - параметр погружения временного ряда (ВР) TS в траекторное пространство
% win  - количество отсчетов ВР TS 
% TS   - ВР, содержащий win отсчетов
   Y = zeros(win-lagS+1,lagS); % траекторная матрица ВР TS
   for m=1:lagS
     Y(:,m) = TS(m:win-lagS+m); % m-й столбец траекторной матрица ВР TS
   end
   Cor = Y'*Y; % lagS*lagS матрица корреляц-х произведений
   lon = lagS;
   CorPro(1:lon) = diag(Cor); % ВР корреляц-го произведения для лага 0 
   Acf(1) = median(CorPro(1:lon)); % медиана главной диагонали CorPro 
   for m=2:lagS
      lon = lon-1;
      CorPro(1:lon) = diag(Cor,m-1); % ВР корреляц-го произведения для лага m-1
      if m<=lagS
         Acf(m) = median(CorPro(1:lon))/Acf(1); % медиан-я оценка нормированной АКФ 
      end
   end
   Acf(1) = 1.0;    
end

File: AcfMed.go
package main

import "gonum.org/v1/gonum/mat"

// Оценка АКФ сингулярных троек для сегментов pw
func ACF_estimation_of_singular_triples(lagS, win, S int, sET12 mat.Dense) mat.Dense {
	//var Acf_sET12 mat.Dense
	Acf_sET12 := mat.NewDense(lagS, S, nil)
	for j := 0; j < S; j++ {
		Acf_sET12.SetCol(j, AcfMed(lagS, win, *mat.VecDenseCopyOf(sET12.ColView(j))))
	}
	return *Acf_sET12
}
func AcfMed(lagS, win int, sET12_vec mat.VecDense) []float64 {
	// lagS - параметр погружения временного ряда (ВР) TS в траекторное пространство
	// win  - количество отсчетов ВР TS
	// TS   - ВР, содержащий win отсчетов

	acf := make([]float64, lagS)

	Y := BuildTrajectoryMatrix222(sET12_vec, lagS, win)

	cor := aTa(Y) // lagS*lagS матрица корреляц-х произведений
	lon := lagS

	CorPro := diag_of_Dense(cor, 0) // ВР корреляц-го произведения для лага 0
	acf[0] = median(CorPro)         // медиана главной диагонали CorPro
	for m := 1; m < lagS; m++ {
		lon--
		diag_cor_minus_1 := diag_of_Dense(cor, m)
		if m < lagS {
			acf[m] = median(diag_cor_minus_1) / acf[0]
		}
	}
	acf[0] = 1.0
	return acf
}
